File: backend/app/schemas/logs.py
from datetime import datetime

from pydantic import BaseModel


class AccessLogItem(BaseModel):
    id: int
    source: str
    path: str
    method: str
    ip: str
    status_code: int
    latency_ms: int
    created_at: datetime


class GiftClaimLogItem(BaseModel):
    id: int
    gift_qrcode_id: int
    red_packet_id: int | None
    dispatch_strategy: str
    ip: str
    result: str
    reason: str
    created_at: datetime


class OperationLogItem(BaseModel):
    id: int
    user_id: int | None
    action: str
    detail: str
    created_at: datetime

File: frontend/src/api/modules/logs.ts
import client from '../client'

interface ApiEnvelope<T> {
  code: string
  message: string
  data: T
}

interface LogListQuery {
  limit?: number
  offset?: number
  q?: string
}

export interface AccessLogItem {
  id: number
  source: string
  path: string
  method: string
  ip: string
  status_code: number
  latency_ms: number
  created_at: string
}

export interface ClaimLogItem {
  id: number
  gift_qrcode_id: number
  red_packet_id: number | null
  dispatch_strategy: string
  ip: string
  result: string
  reason: string
  created_at: string
}

export interface OperationLogItem {
  id: number
  user_id: number | null
  action: string
  detail: string
  created_at: string
}

export async function listAccessLogs(params: LogListQuery = {}): Promise<AccessLogItem[]> {
  const response = await client.get<ApiEnvelope<AccessLogItem[]>>('/logs/access', { params })
  return response.data.data
}

export async function listClaimLogs(params: LogListQuery = {}): Promise<ClaimLogItem[]> {
  const response = await client.get<ApiEnvelope<ClaimLogItem[]>>('/logs/claims', { params })
  return response.data.data
}

export async function listOperationLogs(params: LogListQuery = {}): Promise<OperationLogItem[]> {
  const response = await client.get<ApiEnvelope<OperationLogItem[]>>('/logs/operations', { params })
  return response.data.data
}
